### ISSUE TITLE
Windows: native SSL (Secure Channel) instead of OpenSSL?

### ISSUE THREAD

**Issue by helgeklein** [edenhill/librdkafka#2023]
Description
===========
Our application already uses libcurl for communications with various REST APIs via HTTPS. We would now like to add support for Kafka.

As it seems, SSL support in librdkafka is based on OpenSSL. Since our application already uses the Windows OS native SSL functionality via libcurl it would be quite confusing to our users to configure trusted root certificates and other SSL-related settings in a completely different manner for use with Kafka. Even more importantly, we feel that settings related to security should not be configured individually per application but centrally per OS. Finally, patching of security-related components is much more reliable if done on the OS level.

Long story short: **is there any way to make librdkafka work with the Windows native SSL via the Secure Channel APIs?**

For reference, here is the [SSL library comparison chart](https://curl.haxx.se/docs/ssl-compared.html) from the cURL project.

**Comment by edenhill** [comment]
There is no support, current or planned, for using Windows' native SSL implementation.
But we would be happy to look at a submitted PR!

What is being added is support for pluggable keystores, which allows you to integrate with the Windows SSL key store;
https://github.com/edenhill/librdkafka/pull/1882

**Comment by edenhill** [comment]
Closing in favour of #2309
